### PR TITLE
chore: run hadolint and yamllint during `make lint`

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/1403

yamllint config copied from https://github.com/celestiaorg/.github/blob/main/.yamllint.yml

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,6 +72,8 @@ See <https://docs.celestia.org/category/celestia-app> for more information
 
 1. Install [golangci-lint](https://golangci-lint.run/usage/install/)
 1. Install [markdownlint](https://github.com/DavidAnson/markdownlint)
+1. Install [hadolint](https://github.com/hadolint/hadolint)
+1. Install [yamllint](https://yamllint.readthedocs.io/en/stable/quickstart.html)
```
Comment (cmwaters) [inline README.md:76]: nit: shouldn't the numbers be incrementing (or why are they there)

Comment (cmwaters) [reply README.md:76]: We could also do something like https://github.com/cometbft/cometbft/blob/main/tools/tools.go which, IIUC, will install these tools so developers don't need to manually install them

**Review by cmwaters — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (rootulp) [reply README.md:76]: Markdown will render an ordered list with all leading `1.` with the correct number. This make it easy to re-arrange the lines in the list without manually updating each number.

Ref: https://www.markdownguide.org/basic-syntax/#ordered-lists

The idea from cometbft sounds good to me. Unfortunately in the list above, only golangci-lint is written in Go.

Comment (cmwaters) [reply README.md:76]: > Markdown will render an ordered list with all leading 1. with the correct number. This make it easy to re-arrange the lines in the list without manually updating each number.

TIL

**Pull request merged**